beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -997,6 +997,46 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         Ok(self.store.get_state(state_root, slot)?)
     }
 
+    /// Run a function with mutable access to a state for `block_root`.
+    ///
+    /// The primary purpose of this function is to borrow a state with its tree hash cache
+    /// from the snapshot cache *without moving it*. This means that calls to this function should
+    /// be kept to an absolute minimum, because holding the snapshot cache lock has the ability
+    /// to delay block import.
+    ///
+    /// If there is no appropriate state in the snapshot cache then one will be loaded from disk.
```

Comment:
Just wondering, we prune away old finalized states right? If we do, then this function would return `None` for most of the old states, so we can't compute the old light client updates.
From the [spec](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/full-node.md#create_light_client_update)
> Full nodes SHOULD provide the best derivable LightClientUpdate (according to is_better_update) for each sync committee period covering any epochs in range [max(ALTAIR_FORK_EPOCH, current_epoch - MIN_EPOCHS_FOR_BLOCK_REQUESTS), current_epoch]

Does tree states allow us to have old finalized states too somehow? :o

Reply:
Yeah this is a problem. I started a discussion about it on the R&D server: https://discord.com/channels/595666850260713488/595701319793377299/1040461376126197890

Reply:
So we just maintain a cache of `LightClientUpdates` forward from where we are synced instead of trying to fetch historic updates? Makes sense to me. Feel free to merge this :)